hardware/Honeywell.cpp

```diff
@@ -240,48 +244,91 @@ void CHoneywell::GetThermostatData()
 	}
 
 	int devNr = 0;
-	mDeviceList.clear();
+	mDeviceList.clear(); 
 	mLocationList.clear();
 	for (int locCnt = 0; locCnt < (int)root.size(); locCnt++)
 	{
 		Json::Value location = root[locCnt];
 		Json::Value devices = location["devices"];
 		for (int devCnt = 0; devCnt < (int)devices.size(); devCnt++)
 		{
+			
 			Json::Value device = devices[devCnt];
 			std::string deviceName = device["name"].asString();
 			mDeviceList[devNr] = device;
 			mLocationList[devNr] = location["locationID"].asString();
 
+			std::string units = device["units"].asString(); //:"Fahrenheit"
+
 			float temperature;
 			temperature = (float)device["indoorTemperature"].asFloat();
 			std::string desc = kRoomTempDesc;
 			stdreplace(desc, "[devicename]", deviceName);
-			SendTempSensor(10 * devNr + 1, 255, temperature, desc);
+			if(units == "Fahrenheit") {
+				SendTempSensor(10 * devNr + 1, 255, (temperature - 32) * 5/9, desc);
```

Comment:
Could you use the functions from Helper.h
ConvertToCelsius and if needed ConvertToFahrenheit

Reply:
Thank you for pointing this out, just used them